cpp/src/arrow/filesystem/s3fs_test.cc

```diff
@@ -922,9 +922,13 @@ TEST_F(TestS3FS, CreateDir) {
 
   // New "directory"
   AssertFileInfo(fs_.get(), "bucket/newdir", FileType::NotFound);
-  ASSERT_OK(fs_->CreateDir("bucket/newdir"));
+  ASSERT_OK(fs_->CreateDir("bucket/newdir", false));
```

Comment:
I realize [CreateDir](https://github.com/apache/arrow/blob/main/cpp/src/arrow/filesystem/filesystem.h#L240) defaults to recursive mode. Thus here I'm extending s3fs_test.cc to cover this untested scenario.